src/presentation/http/router/note.test.ts

```diff
@@ -167,6 +174,58 @@ describe('Note API', () => {
       expect(response?.json()).toStrictEqual({ message: 'Note not found' });
     });
 
+    test('Returns public note by public id with status 200 and with false canEdit flag, when user is not authorized', async () => {
+      const expectedStatus = 200;
+      const publicId = 'Pq1T9vc23Q';
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: `/note/${publicId}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json().accessRights).toStrictEqual({ canEdit: false });
+    });
+
+    test('Returns public note by public id with status 200 and with false canEdit flag, when user is authorized, but is not the creator of the note', async () => {
+      const expectedStatus = 200;
+      const publicId = 'Pq1T9vc23Q';
+      const userId = 4;
+      const accessToken = global.auth(userId);
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${publicId}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json().accessRights).toStrictEqual({ canEdit: false });
+    });
+
+    test('Returns public note by public id with status 200 and with false canEdit flag, when user is authorized and is the creator of the note', async () => {
```

Comment:
```suggestion
    test('Returns public note by public id with status 200 and with true canEdit flag, when user is authorized and is the creator of the note', async () => {
```

Reply:
Because in this case accessRights are not settings for a specific note. I was solving a problem when user could open a note and edit it in the note.web (but the changes were not saved in the note.api). To do this, I added sending additional information about user rights to the API. But for now it only checks whether the user is the creator.